server/src/test/java/io/crate/integrationtests/RegexpMatchesTableFunctionIntegrationTest.java

```diff
@@ -0,0 +1,92 @@
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.integrationtests;
+
+import com.carrotsearch.randomizedtesting.annotations.Seed;
+import io.crate.testing.TestingHelpers;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+import static io.crate.testing.TestingHelpers.printedTable;
+import static org.hamcrest.core.Is.is;
+
+public class RegexpMatchesTableFunctionIntegrationTest extends SQLTransportIntegrationTest {
+
+    @Before
+    public void setup() {
+        execute("create table tregex (i integer, s string) with (number_of_replicas=0)");
+        execute("insert into tregex(i, s) values (?, ?)", new Object[][]{
+            new Object[]{1, "foo is first"},
+            new Object[]{2, "bar is second"},
+            new Object[]{3, "foobar is great"},
+            new Object[]{4, "boobar is greater"},
+            new Object[]{5, "awam bam baluma"},
+            new Object[]{6, null}
+        });
+        refresh();
+    }
+
+    @After
+    public void teardown() {
+        execute("drop table tregex");
+    }
+
+    @Seed("C328C6C2B806C0E9")
+    @Test
+    public void test_regexp_matches_is_used_in_select_filtering_from_a_table() {
+        execute("select regexp_matches(s, '(\\w+) is (great).*', 'g') from tregex");
+        Arrays.sort(response.rows(), Comparator.comparing(o -> ((List<String>) o[0]).get(0)));
+        assertThat(TestingHelpers.printedTable(response.rows()), is("[boobar, great]\n" +
+                                                                    "[foobar, great]\n"));
+    }
+
+    @Test
+    public void test_regexp_matches_is_used_in_from_generating_a_table() {
+        execute("select * from regexp_matches('foobar is greater', '(\\w+) is (great|greater)', 'g')");
+        assertThat(TestingHelpers.printedTable(response.rows()), is("[foobar, great]\n"));
+    }
+
+    @Test
+    public void test_regexp_matches_g() {
+        execute("select regexp_matches('foobarbequebazilbarfbonk', '(b[^b]+)(b[^b]+)', 'g')");
+        assertThat(printedTable(response.rows()), is("[bar, beque]\n" +
+                                                     "[bazil, barf]\n"));
+
+        execute("select groups from regexp_matches('foobarbequebazilbarfbonk', '(b[^b]+)(b[^b]+)', 'g')");
+        assertThat(printedTable(response.rows()), is("[bar, beque]\n" +
+                                                     "[bazil, barf]\n"));
+    }
+
+    @Test
+    public void test_regexp_matches() {
+        execute("select regexp_matches('foobarbequebazilbarfbonk', '(b[^b]+)(b[^b]+)')");
+        assertThat(printedTable(response.rows()), is("[bar, beque]\n"));
+
+        execute("select groups from regexp_matches('foobarbequebazilbarfbonk', '(b[^b]+)(b[^b]+)')");
+        assertThat(printedTable(response.rows()), is("[bar, beque]\n"));
+    }
```

Comment:
I don't really see how these are testing anything that is not covered by the unit tests?